--- freetype-2.5.3/builds/unix/freetype-config.in
+++ freetype-2.5.3/builds/unix/freetype-config.in
@@ -114,12 +114,15 @@ if test "$echo_exec_prefix" = "yes" ; th
   echo ${SYSROOT}$exec_prefix
 fi
 
+prefix_var=`pkg-config --variable prefix freetype2`
+lib_suffix=`echo ${libdir} | sed -e 's:'$prefix_var'::'`
+
 if test "$exec_prefix_set" = "yes" ; then
-  libdir=$exec_prefix/lib
+  libdir=${exec_prefix}${lib_suffix}
 else
   if test "$local_prefix" = "yes" ; then
-    includedir=$prefix/include
-    libdir=$prefix/lib
+    includedir=`echo ${includedir} | sed -e 's:'$prefix_var':'${prefix}':'`
+    libdir=${prefix}${lib_suffix}
   fi
 fi
 
