diff -r 59d5dc6a0d95 common/autoconf/hotspot-spec.gmk.in
--- openjdk///common/autoconf/hotspot-spec.gmk.in	Wed May 25 13:42:38 2016 +0100
+++ openjdk///common/autoconf/hotspot-spec.gmk.in	Thu May 26 04:43:57 2016 +0100
@@ -71,6 +71,10 @@
 LIBARCH=$(OPENJDK_TARGET_CPU_LEGACY_LIB)
 # Set the cpu architecture
 ARCH=$(OPENJDK_TARGET_CPU_ARCH)
+# ppc64le uses the HotSpot ppc64 build
+ifeq ($(OPENJDK_TARGET_CPU), ppc64le)
+  ARCH=ppc64
+endif
 # Legacy setting for building for a 64 bit machine.
 # If yes then this expands to _LP64:=1
 @LP64@
diff -r 59d5dc6a0d95 common/autoconf/jdk-options.m4
--- openjdk///common/autoconf/jdk-options.m4	Wed May 25 13:42:38 2016 +0100
+++ openjdk///common/autoconf/jdk-options.m4	Thu May 26 04:43:57 2016 +0100
@@ -158,7 +158,7 @@
   if test "x$JVM_VARIANT_ZEROSHARK" = xtrue ; then
     INCLUDE_SA=false
   fi
-  if test "x$VAR_CPU" = xppc64 ; then
+  if test "x$VAR_CPU" = xppc64 -o "x$VAR_CPU" = xppc64le ; then
     INCLUDE_SA=false
   fi
   if test "x$OPENJDK_TARGET_CPU" = xaarch64; then
diff -r 59d5dc6a0d95 common/autoconf/platform.m4
--- openjdk///common/autoconf/platform.m4	Wed May 25 13:42:38 2016 +0100
+++ openjdk///common/autoconf/platform.m4	Thu May 26 04:43:57 2016 +0100
@@ -67,7 +67,7 @@
       VAR_CPU_ENDIAN=big
       ;;
     powerpc64le)
-      VAR_CPU=ppc64
+      VAR_CPU=ppc64le
       VAR_CPU_ARCH=ppc
       VAR_CPU_BITS=64
       VAR_CPU_ENDIAN=little
diff -r 59d5dc6a0d95 common/autoconf/toolchain.m4
--- openjdk///common/autoconf/toolchain.m4	Wed May 25 13:42:38 2016 +0100
+++ openjdk///common/autoconf/toolchain.m4	Thu May 26 04:43:57 2016 +0100
@@ -1056,6 +1056,9 @@
   else
     CCXXFLAGS_JDK="$CCXXFLAGS_JDK -D_BIG_ENDIAN"
   fi
+  if test "x$OPENJDK_TARGET_CPU" = xppc64le; then
+    CCXXFLAGS_JDK="$CCXXFLAGS_JDK -DABI_ELFv2"
+  fi
   if test "x$OPENJDK_TARGET_OS" = xlinux; then
     CCXXFLAGS_JDK="$CCXXFLAGS_JDK -DLINUX"
   fi
